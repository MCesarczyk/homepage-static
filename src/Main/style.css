.main__wrapper {
  margin: 1.5rem 6rem 2.5rem 6rem;
  border: 1px solid currentColor;
  border-radius: 0.5rem;
  animation: fade-in 2s ease-in-out;
  animation-delay: 0.5s;
  animation-fill-mode: both;
}

@media (max-width: 768px) {
  .main__wrapper {
    margin: 3rem 2rem 5rem 2rem;
  }
}

@keyframes fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.main__buttonsWrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0 -5rem;
}

@media (max-width: 768px) {
  .main__buttonsWrapper {
    flex-direction: column;
    margin: -2.5rem 0;
  }
}

.main__button {
  display: block;
  color: #fff;
  min-height: 1.5rem;
  width: 100%;
  min-width: 75px;
  background-color: #282c34;
  border: 1px solid currentColor;
  border-radius: 4px;
  padding: 0.5rem 1rem;
  margin: 1rem;
  transition: all 0.6s ease-in-out, color 0.1s;
  text-align: center;
  font-size: 1.25rem;
}

@media (max-width: 768px) {
  .main__button {
    width: 50%;
  }
}

.main__button:hover {
  color: #2188ff;
  transform: scale(1.15);
  box-shadow: 0 0 0 5px rgba(31, 135, 255, 0.05), 0 0 0 10px rgba(31, 135, 255, 0.025),
    0 0 0 15px rgba(31, 135, 255, 0.001);
}

.main__button--first:hover {
  transform: scale(1.075) translateX(-0.5rem);
}

.main__button--last:hover {
  transform: scale(1.075) translateX(0.5rem);
}

.main__button:active {
  color: #dc143c;
  box-shadow: 0 0 0 5px rgba(220, 20, 60, 0.05), 0 0 0 10px rgba(220, 20, 60, 0.025),
    0 0 0 15px rgba(220, 20, 60, 0.001);
}

@media (max-width: 768px) {
  .main__button--first:hover {
    transform: scale(1.075) translateY(-0.25rem);
  }

  .main__button--last:hover {
    transform: scale(1.075) translateY(0.25rem);
  }
}
