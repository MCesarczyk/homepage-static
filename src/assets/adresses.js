export const MAIL_ADRESS = "michal.cesarczyk@gmail.com";
export const GITHUB_ADRESS = "https://github.com/MCesarczyk";
export const LINKEDIN_ADRESS = "https://www.linkedin.com/in/micha%C5%82-cesarczyk-0820589a";
export const UPWORK_ADRESS = "https://www.upwork.com/freelancers/~01ec95abd3619f24c6";
export const CODEPEN_ADRESS = "https://codepen.io/bikeoholic";
export const FACEBOOK_ADRESS = "https://www.facebook.com/mich.cesarz/";
export const INSTAGRAM_ADRESS = "https://www.instagram.com/mich_cesarz/";