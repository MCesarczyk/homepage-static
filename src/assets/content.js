export const sections = ["skills", "portfolio", "goals"];