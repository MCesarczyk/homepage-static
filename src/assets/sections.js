export const sections = [
  {
    id: 1,
    name: "skills",
    title: "My skillset",
    previewType: "list",
    items: [
      {
        id: 1,
        name: "Semantic & responsive web design",
        details: [
          "HTML",
          "CSS3",
          "BEM convention",
          "CSS-Flexbox",
          "CSS-Grid"
        ]
      },
      {
        id: 2,
        name: "Javascript",
        details: [
          "JS/ES6+ features",
          "Objects/Arrays operations",
          "Immutability"
        ]
      },
      {
        id: 3,
        name: "Asynchronic operations",
        details: [
          "Promises",
          "Async/Await",
          "AJAX/API/LocalStorage",
          "Error handling"
        ]
      },
      {
        id: 4,
        name: "React",
        details: [
          "React hooks/custom Hooks",
          "React-Redux/Redux-toolkit",
          "Redux-Saga",
          "React Router"
        ]
      },
      {
        id: 5,
        name: "React app styling",
        details: [
          "Styled components",
          "Ant Design UI library",
          "Material UI library",
          "Bootstrap"
        ]
      },
      {
        id: 6,
        name: "Building fulstack apps",
        details: [
          "Ruby on Rails framework",
          "PostgreSQL basics",
          "Restful API",
          "packet managers: NVM, RVM, Gem bundler"
        ]
      },
      {
        id: 7,
        name: "Deploying apps",
        details: [
          "Git/gh-pages",
          "Digitalocean",
          "Nginx/Passenger",
          "AWS S3 storage handling"
        ]
      },
      {
        id: 8,
        name: "Teamworking",
        details: [
          "GitHub flow, collaborating",
          "Trello: tasks management",
          "Figma: reading UI layout",
          "Scrum/Agile metodology"
        ]
      },
      {
        id: 9,
        name: "Other skills",
        details: [
          "Expertise in CNC programming",
          "CNC machines setting/operation",
          "General technical knowledge"
        ]
      }
    ]
  },
  {
    id: 2,
    name: "portfolio",
    title: "My portfolio",
    previewType: "gallery",
    items: []
  },
  {
    id: 3,
    name: "goals",
    title: "My closest goals",
    previewType: "list",
    items: [
      {
        id: 1,
        name: "Things I'm currently learning",
        details: [
          "Testing code: Jest/Enzyme",
          "TypeScript"
        ]
      },
      {
        id: 2,
        name: "Things I plan to learn subsequently",
        details: [
          "Node.js",
          "GraphQL",
          "Next.js"
        ]
      }
    ]
  }
];