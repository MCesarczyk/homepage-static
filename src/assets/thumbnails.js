import { ReactComponent as GithubThumbnail } from "./githubThumbnail.svg";
import { ReactComponent as LinkedinThumbnail } from "./linkedinThumbnail.svg";
import { ReactComponent as UpworkThumbnail } from "./upworkThumbnail.svg";

import {
    GITHUB_ADRESS,
    LINKEDIN_ADRESS,
    UPWORK_ADRESS,
} from "./adresses";

export const thumbnails = [
    {
        id: 1,
        testId: "github-icon-test",
        icon: <GithubThumbnail />,
        url: GITHUB_ADRESS,
    },
    {
        id: 2,
        testId: "linkedin-icon-test",
        icon: <LinkedinThumbnail />,
        url: LINKEDIN_ADRESS,
    },
    {
        id: 3,
        testId: "upwork-icon-test",
        icon: <UpworkThumbnail />,
        url: UPWORK_ADRESS,
    },
];