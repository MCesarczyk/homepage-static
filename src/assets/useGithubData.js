import { useEffect, useState } from "react";
import { fetchAPIData } from "./fetchAPIData";

export const useGithubData = () => {
  const [repos, setRepos] = useState({
    status: "loading",
    repos: null
  });

  const apiURL = 'https://api.github.com/users/MCesarczyk/repos?sort="updated"';

  useEffect(() => {
    const getRepos = async () => {
      try {
        const repos = await fetchAPIData(apiURL);

        if (!repos) {
          throw new Error("No repo data");
        }

        setRepos({
          status: "success",
          repos: repos
        });

      } catch (error) {
        setRepos({ status: "error" });
        console.log(error.message);
      }
    };

    setTimeout(getRepos, 2_000);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return repos;
};