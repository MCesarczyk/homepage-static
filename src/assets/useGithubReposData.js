import { useEffect, useState } from "react";
import { fetchAPIData } from "./fetchAPIData";

export const useGithubReposData = () => {
  const [repos, setRepos] = useState({
    status: "loading",
    repos: []
  });

  const DEMO_DELAY = 2_500;

  const apiURL = 'https://api.github.com/users/MCesarczyk/repos?sort="updated"';

  useEffect(() => {
    const getRepos = async () => {
      try {
        const repos = await fetchAPIData(apiURL);

        if (!repos) {
          throw new Error("No repo data");
        }

        setRepos({
          status: "success",
          repos: repos
        });

      } catch (error) {
        setRepos({ status: "error" });
        console.error(error.message);
      }
    };

    setTimeout(getRepos, DEMO_DELAY);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return repos;
};