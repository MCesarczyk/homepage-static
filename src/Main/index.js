import { sections } from "../assets/sections";
import "./style.css";

const Main = ({ openModal }) => (
  <main className="main__wrapper">
    <div className="main__buttonsWrapper">
      {sections.map((section, index) =>
        <button
          key={index}
          className={`
                main__button
                ${index === 0 && "main__button--first"}
                ${index === sections.length - 1 && "main__button--last"}
              `}
          onClick={() => openModal(section)}
        >
          {section.name}
        </button>
      )}
    </div>
  </main>
);

export default Main;