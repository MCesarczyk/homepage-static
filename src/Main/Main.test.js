import { render, screen } from '@testing-library/react';
import Main from '.';
import { sections } from '../assets/content';

test('renders sections names', () => {
  render(<Main />);

  sections.forEach(section => {
    const title = screen.getByText(section);
    expect(title).toBeInTheDocument();
  });
});

test('buttons are rendering properly', () => {
  render(<Main />);
  const buttons = screen.getAllByRole('button');

  buttons.forEach(button => {
    expect(button).toMatchSnapshot();
  });
});

test('buttons have basic style', () => {
  render(<Main />);
  const buttons = screen.getAllByRole('button');

  buttons.forEach(button => {
    expect(button).toHaveStyle("css:", "main__button");
  });
});

test('first and last buttons have additional styles', () => {
  render(<Main />);
  const buttons = screen.getAllByRole('button');
  const first = buttons[0];
  const last = buttons[buttons.length - 1];

  expect(first).toHaveStyle("css:", "main__button--first");
  expect(last).toHaveStyle("css:", "main__button--last");
});