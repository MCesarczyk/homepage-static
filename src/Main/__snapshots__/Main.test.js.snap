// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttons are rendering properly 1`] = `
<button
  class="
                main__button
                main__button--first
                false
              "
>
  skills
</button>
`;

exports[`buttons are rendering properly 2`] = `
<button
  class="
                main__button
                false
                false
              "
>
  portfolio
</button>
`;

exports[`buttons are rendering properly 3`] = `
<button
  class="
                main__button
                false
                main__button--last
              "
>
  goals
</button>
`;
